src/Acuminator/Acuminator.Analyzers/Analyzers/Localization/LocalizationMessageHelper.cs

```diff
@@ -0,0 +1,136 @@
+            {
+                if (IsNonLocalizableMessageType(messageType))
+                {
+                    _syntaxContext.ReportDiagnostic(Diagnostic.Create(Descriptors.PX1051_NonLocalizableString, _messageExpression.GetLocation()));
+                }
+
+                if (IsIncorrectStringToFormat())
+                {
+                    _syntaxContext.ReportDiagnostic(Diagnostic.Create(Descriptors.PX1052_IncorrectStringToFormat, _messageExpression.GetLocation()));
+                }
+            }
+            else if (IsStringConcatenation())
+            {
+                _syntaxContext.ReportDiagnostic(Diagnostic.Create(Descriptors.PX1053_ConcatinationPriorLocalization, _messageExpression.GetLocation()));
+            }
+        }
+
+        private ITypeSymbol ReadMessageInfo()
+        {
+            if (Cancellation.IsCancellationRequested)
+                return null;
+
+            if (!(_messageExpression is MemberAccessExpressionSyntax memberAccess) ||
+                !(memberAccess.Name is IdentifierNameSyntax messageMember))
+                return null;
+
+            ITypeSymbol messageClassType = memberAccess.Expression
+                                           .DescendantNodesAndSelf()
+                                           .OfType<IdentifierNameSyntax>()
+                                           .Select(i => SemanticModel.GetTypeInfo(i, Cancellation).Type)
+                                           .Where(t => t != null && t.IsReferenceType)
+                                           .FirstOrDefault();
+            if (messageClassType == null)
+                return null;
+
+            ISymbol messageMemberInfo = SemanticModel.GetSymbolInfo(messageMember, Cancellation).Symbol;
+            if (messageMemberInfo == null || messageMemberInfo.Kind != SymbolKind.Field || !messageMemberInfo.IsStatic ||
+                messageMemberInfo.DeclaredAccessibility != Accessibility.Public)
+                return null;
+
+            _messageMember = messageMemberInfo;
+            return messageClassType;
+        }
+
+        private bool IsNonLocalizableMessageType(ITypeSymbol messageType)
+        {
+            if (Cancellation.IsCancellationRequested)
+                return false;
+
+            ImmutableArray<AttributeData> attributes = messageType.GetAttributes();
+            return attributes.All(a => !a.AttributeClass.Equals(Localization.PXLocalizableAttribute));
+        }
+
+        private bool IsIncorrectStringToFormat()
+        {
+            if (!_isFormatMethod || Cancellation.IsCancellationRequested)
+                return false;
+
+            Optional<object> constString = SemanticModel.GetConstantValue(_messageExpression);
+            if (!constString.HasValue)
+                return false;
+
+            Regex regex = new Regex(_formatRegExp, RegexOptions.CultureInvariant);
+            MatchCollection matchesValue = regex.Matches(constString.Value as string);
```

Comment:
+1, this Regex should be compiled and static

Reply:
@metallmonster I see, in this case it's okay. Maybe the Jira item should be added or TODO comment in the code - just in case, to write down this good idea somewhere.

Reply:
@SENya1990 I've created Jira item ATR-203 for that.

Reply:
@metallmonster thank you